### PR TITLE
New package: py-correctionlib

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-correctionlib/package.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    pypi     = "correctionlib/correctionlib-2.0.0.tar.gz"
+
+    version('2.0.0', sha256='e4d240cbdb2633a8955ddcd02d5b9bfb33d7e1a33554d6f7957f2dec56988a67')
+
+    variant('convert', default=False,
+            description='Includes select conversion routines for common types')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-correctionlib/package.py]:
```suggestion
    depends_on('py-setuptools@42:', type='build')
    depends_on('py-setuptools-scm@3.4:+toml', type='build')
```
From `pyproject.toml`

File: var/spack/repos/builtin/packages/py-correctionlib/package.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on('py-scikit-build', type='build')
+    depends_on('cmake', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-correctionlib/package.py]:
```suggestion
    depends_on('py-cmake@3.11:', type='build')
```

File: var/spack/repos/builtin/packages/py-correctionlib/package.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on('ninja', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-correctionlib/package.py]: Are you sure it uses ninja? The `pyproject.toml` explicitly lists `py-make`.

File: var/spack/repos/builtin/packages/py-correctionlib/package.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on('py-numpy@1.13.3:', type=('build', 'run'))
+    depends_on('py-typing', type=('build', 'run'), when='^python@:3.4')
+    depends_on('py-typing-extensions', type=('build', 'run'), when='^python@:3.7')
+    depends_on('py-dataclasses', type=('build', 'run'), when='^python@:3.6')
+    depends_on('py-pydantic@1.7.3:', type=('build', 'run'))
+    depends_on('py-rich', type=('build', 'run'))
+    depends_on('py-pandas', type=('build', 'run'), when='+convert')
+    depends_on('py-uproot@4.0.4:', type=('build', 'run'), when='+convert')
+    depends_on('py-requests', type=('build', 'run'), when='+convert')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-correctionlib/package.py:35]: Missing build dep on `py-pybind11@2.6.1:`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-correctionlib/package.py (hunk 3 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-correctionlib/package.py:25]: Does it?
> `requires = ["setuptools>=42", "wheel", "cmake>=3.18", "ninja"]`

The log indicates that it tries Ninja first, falling back to GNU Make if not available.

**Comment by iarspider** [comment]
@adamjstewart  This is a very confusing package. 

One, pybind11, rapidjson and cpp-peglib, are bundled, but still pybind11 is listed as a dependency. 

Two, in cmsdist (which build the package using modern approach - `pip install`), the requirements are listed as 

```
Requires: py3-pybind11 py3-numpy py3-pydantic py3-python-rapidjson py3-rich
BuildRequires: py3-setuptools-scm py3-scikit-build cmake 
```

(notice: not py3-cmake, but cmake). And it works, it passes tests, etc...

I guess I should implement the recipe as written (i.e. add `py-make` recipe, add `py-bind11` dependency, etc)?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**